File: src/main/java/com/koddy/server/global/exception/ExceptionResponse.java
package com.koddy.server.global.exception;

import com.koddy.server.global.base.BaseExceptionCode;

public record ExceptionResponse(
        String errorCode,
        String message
) {
    public ExceptionResponse(final BaseExceptionCode code) {
        this(code.getErrorCode(), code.getMessage());
    }

    public ExceptionResponse(final BaseExceptionCode code, final String message) {
        this(code.getErrorCode(), message);
    }
}

File: src/main/kotlin/com/koddy/server/global/exception/ExceptionResponse.kt
package com.koddy.server.global.exception

import com.koddy.server.global.base.BusinessExceptionCode

data class ExceptionResponse(
    val errorCode: String,
    val message: String,
) {
    constructor(code: BusinessExceptionCode) : this(
        errorCode = code.errorCode,
        message = code.message,
    )

    constructor(
        code: BusinessExceptionCode,
        message: String,
    ) : this(
        errorCode = code.errorCode,
        message = message,
    )
}
